Rubberduck.VBEEditor/Extensions/CodeModuleExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Text.RegularExpressions;
+using Rubberduck.VBEditor.SafeComWrappers.Abstract;
+
+namespace Rubberduck.VBEditor.Extensions
+{
+    public static class CodeModuleExtensions
+    {
+        public static bool ContainsCompilationDirectives(this ICodeModule codeModule, Selection selection)
+        {
+            // We need to inspect the text directly from the code module because the parse tree do not have compilation
+            // constants or directives easily available (they are in a hidden channel and are encapsulated). Fortunately,
+            // finding compilation constants or directives are easy; they must be prefixed by a "#" and can only have
+            // whitespace between it and the start of the line. Labels or line numbers are not legal in those lines.
+            var rawCode = string.Join(Environment.NewLine,
+                codeModule.GetLines(selection));
+            var regex = new Regex(@"^(\s?)+#", RegexOptions.Multiline);
+            return regex.Matches(rawCode).Count > 0;
+        }
+
+        public static bool ContainsCompilationDirectives(this ICodeModule codeModule)
+        {
+            var rawCode = string.Join(Environment.NewLine,
+                codeModule.GetLines(1, codeModule.CountOfLines));
+            var regex = new Regex(@"^(\s?)+#", RegexOptions.Multiline);
+            return (regex.Matches(rawCode).Count > 0);
+        }
```

Comment:
I think I'd rewrite this to call something like:

```
private static bool ContainsCompilationDirectives(string rawCode) 
{
    // local static may not be supported, if it isn't extract it into a constant
    static var regex = new Regex(@"^\s*#", RegexOptions.Multiline);
    return regex.Matches(rawCode).Count > 0;
}
```

and just call this from the other extension methods